express/component_spec.py

```diff
@@ -0,0 +1,116 @@
+"""This module defines classes to represent an Express component specifications."""
+import json
+import pkgutil
+import yaml
+import typing as t
+from typing import Dict
+from dataclasses import dataclass
+
+import jsonschema.exceptions
+from jsonschema import Draft4Validator
+
+from express.exceptions import InvalidComponentSpec
+
+
+def load_yaml(yaml_path: str) -> Dict:
+    """Loads a YAML file and returns a dictionary."""
+    try:
+        with open(yaml_path) as f:
+            return yaml.safe_load(f)
+    except Exception as e:
+        raise ValueError(f"Failed to load YAML file: {e}")
+
+
+@dataclass
+class ComponentField:
+    """Class representing a single field or column in an Express component subset."""
+
+    name: str
+    type: str
+
+
+@dataclass
+class ComponentSubsets:
+    """
+    Class representing an Express Component subset.
+    Args:
+        specification: A dictionary representing the component input and output subset specs
+    """
+
+    def __init__(self, specification: dict) -> None:
+        """
+        Class representing a component subset
+        Args:
+            specification: the part of the component json representing the subset
+        """
+        self._specification = specification
+
+    @property
+    def fields(self) -> t.Dict[str, ComponentField]:
+        print(self._specification)
+        return {
+            name: ComponentField(name=name, type=field["type"])
+            for name, field in self._specification["fields"].items()
+        }
+
+
+class ExpressComponent:
+    """
+    Class representing an Express component
+    Args:
+        specification: The component specification as a Python dict
```

Comment:
The yaml_spec_path can be added to the docstring

Reply:
I'm also a fan of somehow merging the 2 specs. I would say using the Fondant spec to generate the kubeflow spec (while compiling the pipeline). The current kubeflow spec seems very minimal.

Reply:
I think this would be a good idea eventually to have a single reference point for defining the components (right now you have to define both a component spec class + component yaml file). We can expand the ComponentSpec class to generate the yaml file but best to tackle this in a separate PR. I'll create a ticket for it.